experimental/traffic-portal/src/app/api/testing/physical-location.service.ts

```diff
@@ -12,56 +12,105 @@
+	public async getPhysicalLocations(nameOrID: string | number): Promise<ResponsePhysicalLocation>;
 
-	public async getPhysicalLocations(idOrName: number | string): Promise<PhysicalLocation>;
-	public async getPhysicalLocations(): Promise<Array<PhysicalLocation>>;
 	/**
-	 * Gets one or all PhysicalLocations from Traffic Ops
+	 * Gets one or all of PhysicalLocations from Traffic Ops.
 	 *
-	 * @param idOrName Either the integral, unique identifier (number) or name (string) of a single PhysicalLocation to be returned.
-	 * @returns The requested PhysicalLocation(s).
+	 * @param nameOrID If given, returns only the ResponsePhysicalLocation with the given name
+	 * (string) or ID (number).
+	 * @returns An Array of ResponsePhysicalLocation objects - or a single ResponsePhysicalLocation object if 'nameOrID'
+	 * was given.
 	 */
-	public async getPhysicalLocations(idOrName?: number | string): Promise<PhysicalLocation | Array<PhysicalLocation>> {
-		if (idOrName !== undefined) {
-			let loc;
-			switch (typeof idOrName) {
+	public async getPhysicalLocations(nameOrID?: string | number): Promise<Array<ResponsePhysicalLocation> | ResponsePhysicalLocation> {
+		if(nameOrID) {
+			let physicalLocation;
+			switch (typeof nameOrID) {
 				case "string":
-					loc = this.locs.find(l=>l.name === idOrName);
+					physicalLocation = this.physicalLocations.find(d=>d.name === nameOrID);
 					break;
 				case "number":
-					loc = this.locs.find(l=>l.id === idOrName);
+					physicalLocation = this.physicalLocations.find(d=>d.id === nameOrID);
 			}
-			if (!loc) {
-				throw new Error(`no such Physical Location: ${idOrName}`);
+			if (!physicalLocation) {
+				throw new Error(`no such PhysicalLocation: ${nameOrID}`);
 			}
-			return loc;
+			return physicalLocation;
+		}
+		return this.physicalLocations;
+	}
+
+	/**
+	 * Replaces the current definition of a physicalLocation with the one given.
+	 *
+	 * @param physicalLocation The new physicalLocation.
+	 * @returns The updated physicalLocation.
+	 */
+	public async updatePhysicalLocation(physicalLocation: ResponsePhysicalLocation): Promise<ResponsePhysicalLocation> {
+		const id = this.physicalLocations.findIndex(d => d.id === physicalLocation.id);
+		if (id === -1) {
+			throw new Error(`no such PhysicalLocation: ${physicalLocation.id}`);
+		}
+		this.physicalLocations[id] = physicalLocation;
+		return physicalLocation;
+	}
+
+	/**
+	 * Creates a new physicalLocation.
+	 *
+	 * @param physicalLocation The physicalLocation to create.
+	 * @returns The created physicalLocation.
+	 */
+	public async createPhysicalLocation(physicalLocation: RequestPhysicalLocation): Promise<ResponsePhysicalLocation> {
+		return {
+			...physicalLocation,
+			comments: physicalLocation.comments ?? null,
+			email: physicalLocation.email ?? null,
+			id: ++this.lastID,
+			lastUpdated: new Date(),
+			phone: physicalLocation.phone ?? null,
+			poc: physicalLocation.poc ?? null,
+			region: ""
+		};
+	}
```

Comment:
This doesn't appear to actually be inserting the Physical Location into the testing data